semantics/src/main/kotlin/com/strumenta/kolasu/semantics/identifier/provider/IdentifierProvider.kt

```diff
@@ -0,0 +1,15 @@
+package com.strumenta.kolasu.semantics.identifier.provider
+
+import com.strumenta.kolasu.model.Node
+import kotlin.reflect.KClass
+
+interface IdentifierProvider {
+    /**
+     * Retrieve the identifier associated
+     * with the given node (possibly cast as requested)
+     **/
+    fun <NodeTy : Node> getIdentifierFor(
+        node: NodeTy,
```

Comment:
I would add a comment to explain why someone may want to pass `typedAs`

Reply:
I agree, the current implementation is not really well documented as the main focus consisted in experimenting first. I will add a detailed explanation of every component once done, thanks.